processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -660,35 +665,33 @@ public DateTime getMaxIngestedEventTime()
 
   class DimensionHolder
   {
-    private final Map<String, DimDim> dimensions;
+    private final DimDim[] dimensions;
 
-    DimensionHolder()
+    DimensionHolder(int maxDimensions)
```

Comment:
what happens when we exceed maxDimensions? How will it work with schema-less indexing?